aten/src/ATen/native/cpu/UpSampleKernel.cpp

```diff
@@ -1340,8 +1350,12 @@ void upsample_bilinear2d_kernel_impl(
     c10::optional<double> scales_h,
     c10::optional<double> scales_w) {
 
-  // Temporarily dispatch to original channels last implementation
-  if (input.is_contiguous(at::MemoryFormat::ChannelsLast)) {
+  // See note above about USE_VECTORIZED_KERNEL_COND. The extra cond is present
+  // because benchmarks showed that with only 1 thread, images (C == 3) were
+  // slightly faster with the vectorized kernel than with the generic one.
+  // That's not the case for masks though (C == 1), which strongly benefit from
+  // using the generic kernel.
```

Comment:
This is probably because for C = 1 in the channels-last case, the generic implementation can generate vectorized code right away